### PR TITLE
Ban emojis in post titles

### PR DESCRIPTION
**Pull request by oetherington**
You don't get to use that regex every day. It's a fun one.

Screenshot 2023-05-18 at 23 56 08

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1204633037683998) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by Raemon** [comment]
I think in my heart-of-hearts I want each user to get to do this, like, exactly once, and only if they have like 1000 karma. But this sure seems easier.

**Comment by b0b3rt** [comment]
Did you know that we permit the zero-width space (char code 8203) as the entire contents of a post title, which (at least on LessWrong) makes it impossible to click on the post title?

**Comment by jpaddison3** [comment]
> Did you know that we permit the zero-width space (char code 8203) as the entire contents of a post title, which (at least on LessWrong) makes it impossible to click on the post title?

This comment enriched my morning

File: packages/lesswrong/server/callbacks/postCallbacks.ts [hunk 1]
```diff
@@ -41,6 +41,14 @@ if (forumTypeSetting.get() === "EAForum") {
   getCollectionHooks("Posts").updateBefore.add(
     (post, {currentUser}) => checkTosAccepted(currentUser, post),
   );
+
+  const assertPostTitleHasNoEmojis = (post: DbPost) => {
+    if (/\p{Extended_Pictographic}/u.test(post.title)) {
```
Comment (jpaddison3) [inline packages/lesswrong/server/callbacks/postCallbacks.ts:46]: ..huh, neat

**Review by jpaddison3 — APPROVED** [review]
I assume that you double checked that posts without emojis can still be submitted.

\@LW It sounds like you want this, but I am not >80% confident that you do, and so my recommendation is that we ship this and y'all can refactor later if you want it.

File: packages/lesswrong/server/callbacks/postCallbacks.ts [hunk 2]
```diff
@@ -41,6 +41,14 @@ if (forumTypeSetting.get() === "EAForum") {
...
+      throw new Error("Post titles cannot contain emojis");
```
[the remark this answers is not in the document]
Comment (oetherington) [reply packages/lesswrong/server/callbacks/postCallbacks.ts:47]: Ha! That's fair - I feel like there should at least be an emoji in the error message. Maybe a question for the commons on slack?

[the remark this answers is not in the document]
Comment (jpaddison3) [reply packages/lesswrong/server/callbacks/postCallbacks.ts:47]: :D

**Pull request merged**